### DISCUSSION TITLE
Error while using '@tailwind base'

### DISCUSSION THREAD

**Discussion by parikmaan** [tailwindlabs/tailwindcss#7635] [Help]
### Discussed in https://github.com/tailwindlabs/tailwindcss/discussions/7628

<div type='discussions-op-text'>

<sup>Originally posted by **parikmaan** February 23, 2022</sup>
Hi

I have setup a basic angular project. Some of the styles are not matching what I see on the website. I am getting following error in browser console:

```
Error while fetching an original source: NetworkError when attempting to fetch resource.
Source URL: webpack:///%3Cno%20source%3E
```

<img width="1680" alt="Screen Shot 2022-02-23 at 10 26 29 PM" src="https://user-images.githubusercontent.com/25786988/155452317-aa729f69-70cf-4ed3-9467-cfe92deefe1b.png">

Error occurs when I include any one of these in my src/styles.css file:

```
@tailwind base;
@tailwind utilities;
```

but `@tailwind components;` works fine.

Versions used:

```
Angular CLI: 13.2.5
Node: 16.13.2
Package Manager: npm 8.1.2
```

Tailwind CSS dependencies:

```
"postcss": "^8.4.6",
"tailwindcss": "^3.0.23",
"autoprefixer": "^10.4.2",
```

styles.css:

```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

html file:

`<input id="LoggingEmailAddress" class="block w-full px-4 py-2 text-gray-700 bg-white border rounded-md dark:bg-gray-800 dark:text-gray-300 dark:border-gray-600 focus:border-blue-400 focus:ring-opacity-40 dark:focus:border-blue-300 focus:outline-none focus:ring focus:ring-blue-300" type="email">`

</div>

**Comment by simonswiss** [comment]
Hey - are you able to share a reproduction of the problem in a GitHub repo?

Happy to have a look, but I need to be able to play with the code for that 👍

**Reply by parikmaan** [reply]
Hi, here it is. 

https://github.com/parikmaan/sample-ui-tailwindcss

Not sure what is causing this issue. My suspicion is for css processors or something like that. But not able to figure it out.

**Comment by NovusOpus** [comment]
Your `purge`/`content` configuration is invalid. [In Tailwind CSS V3 the `purge` option was replaced with `content` since Tailwind no longer uses PurgeCSS](https://tailwindcss.com/docs/upgrade-guide#configure-content-sources).

Currently you have this:
```JS
module.exports = {
  prefix: '',
  mode: 'jit',
  purge: {
    enabled: true,
    content: [
      './src/**/*.{html,ts}',
    ]
  },
  darkMode: 'media', // or 'media' or 'class'
  theme: {
    fontFamily: {
      sans: ['Helvetica Neue', 'Helvetica','Arial','sans-serif']
    },
    extend: {
      colors: {
        'picton-blue': {
          light: '#337AB7',
          DEFAULT: '#337AB7',
          dark: '#14b367',
        },
        'nav-hover-blue': {
          DEFAULT: '#286090',
        },
        'nav-menu-active':{
          DEFAULT: '#1d4568',
        },
        'new': {
          DEFAULT: '#D9EDF7',
        },
      }
    },
  },
  corePlugins: {
    preflight: true,
  },
  variants: {
    extend: {},
  },
  plugins: [require('@tailwindcss/forms'),require('@tailwindcss/typography')],
};
```
This is the correct configuration with V3:
```JS
module.exports = {
  content: [
    './src/**/*.{html,ts}',
  ],
  theme: {
    fontFamily: {
      sans: ['Helvetica Neue', 'Helvetica','Arial','sans-serif']
    },
    extend: {
      colors: {
        'picton-blue': {
          light: '#337AB7',
          DEFAULT: '#337AB7',
          dark: '#14b367',
        },
        'nav-hover-blue': {
          DEFAULT: '#286090',
        },
        'nav-menu-active':{
          DEFAULT: '#1d4568',
        },
        'new': {
          DEFAULT: '#D9EDF7',
        },
      }
    },
  },
  plugins: [require('@tailwindcss/forms'),require('@tailwindcss/typography')],
};
```
Note that I did more changes to your tailwind.config.js file and they listed here:

1. [With Tailwind CSS V3+ You don't need to use: `mode: 'jit'` since now the JIT Engine is the enabled by default and has replaced the classic engine](https://tailwindcss.com/docs/upgrade-guide#migrating-to-the-jit-engine).

2. [You can also remove `darkMode: 'media'`](https://tailwindcss.com/docs/upgrade-guide#remove-dark-mode-configuration)

3. [Also you can remove the `variants` option, Since in Tailwind CSS v3.0, every variant is automatically available for every utility by default](https://tailwindcss.com/docs/upgrade-guide#remove-variant-configuration).

4. [Remove: `corePlugins: {preflight: true}`, You only need to add this if the value is false `corePlugins: {preflight: false}`](https://tailwindcss.com/docs/preflight#extending-preflight)


Now, try addressing this and see if the problem has been solved?

**Reply by parikmaan** [reply]
Thanks, it was very detailed and helpful explanation. I tried these changes but still see same error.

<img width="1680" alt="Screen Shot 2022-02-25 at 9 27 35 AM" src="https://user-images.githubusercontent.com/25786988/155731937-895ce4b3-8085-49c5-9fdb-6f4ebe00b10c.png">

**Comment by simonswiss** [comment]
I pulled the repo and things look pretty good to me! Can you be a bit more specific on what styles are missing?

![image](https://user-images.githubusercontent.com/485747/155659082-a8bbbff3-cf83-42f3-8c7a-4f56c7118e47.png)

The old `purge.content` configuration should still work properly in v3, we made it backward compatible.

That said, @parikmaan, all the suggestions made above by @NovusOpus are good, I recommend you make these changes 👍

Also, I am pretty sure the webpack warning you're getting in the console has nothing to do with Tailwind CSS.

**Reply by parikmaan** [reply]
@simonswiss this is the expected UI when using CSS:

<img width="923" alt="Screen Shot 2022-02-25 at 9 29 15 AM" src="https://user-images.githubusercontent.com/25786988/155732199-133ae201-6f02-41c6-b2a2-85e1d7489e02.png">

If you look at the input boxes and border of sign in with google. They are not same as expected UI template. I have used exactly same code.

After further debugging I found that in the UI template I am using the code from has few styles coming in. Highlighted here:

<img width="1680" alt="Screen Shot 2022-02-25 at 9 31 11 AM" src="https://user-images.githubusercontent.com/25786988/155733059-cca3b8a3-73be-40fd-b55b-5d4effa40248.png">

<img width="1680" alt="Screen Shot 2022-02-25 at 9 32 13 AM" src="https://user-images.githubusercontent.com/25786988/155733081-cd85c158-33ca-4e9a-a353-6f768ab8432c.png">

<img width="1680" alt="Screen Shot 2022-02-25 at 9 33 12 AM" src="https://user-images.githubusercontent.com/25786988/155733095-801a8134-b8c9-4b16-8e06-60c70c0e5a14.png">

But in my project this CSS is not loading. Here are screenshots:

<img width="1680" alt="Screen Shot 2022-02-25 at 9 36 53 AM" src="https://user-images.githubusercontent.com/25786988/155733558-40708c62-7cfb-4eee-a3fb-b7e77237bf7d.png">

<img width="1680" alt="Screen Shot 2022-02-25 at 9 37 59 AM" src="https://user-images.githubusercontent.com/25786988/155733985-d1106610-23c3-409f-bab3-084ad4ccf88e.png">

Not sure what is the problem here. Why these styles are not loading? I do see base.css is there but they are not getting loaded.
